src/app/admin/modals/user-settings-modal/user-settings-modal.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { currentUser } from 'src/app/ajs-upgraded-providers';
+import { User, UserService } from 'src/app/api/models/doubtfire-model';
+import { DoubtfireConstants } from 'src/app/config/constants/doubtfire-constants';
+
+@Component({
+  selector: 'user-settings-modal',
+  templateUrl: 'user-settings-modal.component.html',
+  styleUrls: ['user-settings-modal.component.scss'],
+})
+export class UserSettingsModalComponent implements OnInit {
+  // create or not
+  isNew: boolean = true;
+  // user information list
+  user: User | null;
+
+  externalName: string;
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    @Inject(currentUser) public CurrentUser: any,
+    public dialogRef: MatDialogRef<UserSettingsModalComponent>,
+    private constants: DoubtfireConstants,
+    private userService: UserService
+  ) {}
+
+  ngOnInit(): void {
+    this.constants.ExternalName.subscribe((result) => {
+      this.externalName = result;
+    })
```

Comment:
**Question**: Do we need `subscribe`? Isn't the ` ExternalName ` directly from ` DoubtfireConstants.ExternalName `?